### PR TITLE
Refacto, adds new /prestashop endpoints and delete module download endpoint

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | This PR aims to refactor the code, by relying more on objects instead of arrays. It also add the endpoint `/prestashop` & `/prestashop/<channel>` which returns every PrestaShop versions and the latest PrestaShop version of the specified channel respectively. Module download endpoint has been removed.
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | CI Should be green
| Possible impacts? |

### CODE REVIEW

File: src/Command/CleanCommand.php [hunk 1]
```diff
@@ -13,25 +13,25 @@
...
 
     private Filesystem $filesystem;
     private string $moduleDir;
+    private string $prestaShopDir;
     private string $jsonDir;
-    private string $tmpDir;
 
     public function __construct(
-        string $moduleDir = __DIR__ . '/../../public/modules',
+        string $moduleDir = __DIR__ . '/../../var/tmp/modules',
```
Comment (PierreRambaud) [inline src/Command/CleanCommand.php:26]: You can inject the `%kernel.project_dir%` into this command instead

File: config/services.yaml [hunk 2]
```diff
@@ -5,6 +5,10 @@ services:
   _defaults:
     public: true
     autowire: true
+    bind:
+      $jsonDir: '%kernel.project_dir%/public/json'
+      $moduleDir: '%kernel.project_dir%/var/tmp/modules'
```
Comment (PierreRambaud) [inline config/services.yaml:10]: Are they really temporary files?

Comment (PierreRambaud) [reply config/services.yaml:10]: (just a question ^^)

Comment (atomiix) [reply config/services.yaml:10]: Mostly yeah, it can be seen as cache, to retrieve modules/PrestaShop information needed to generate the json files.

Comment (PierreRambaud) [reply config/services.yaml:10]:
```suggestion
      $moduleDir: '%kernel.project_dir%/var/cache/modules'
```

maybe use var/cache instead if it's here for cache purposes.

**Review by PierreRambaud — APPROVED** [review]
Let's merge this, and refacto if needed depending on how we will integrate the API inside the Core project.

**Pull request merged**